### PR TITLE
Implement CASE WHEN format

### LINKED ISSUE #36: Linebreak CASE… WHEN
See https://github.com/jdorn/sql-formatter/issues/83

### PR DESCRIPTION
**Pull request by mvorisek**
closes #36

Supporting CASE WHEN SQL is important to format queries like https://github.com/atk4/data/blob/a9c6063420/tests/Persistence/Sql/QueryTest.php#L867 to be readable.

Please merge into 1.3.x as we need PHP 7.4 support.

### CODE REVIEW

**Comment by mvorisek** [comment]
Can CI for me in this repo be approved and can I have a review on this PR, thank you.

**Comment by derrabus** [comment]
> Please merge into 1.3.x as we need PHP 7.4 support.

We don't maintain 1.3 anymore. Please target 1.4.x.

**Comment by mvorisek** [comment]
Would it be possible to do 1 release with this PR for PHP 7.4 based on 1.3.x branch?

I ask for this as in atk4/data I want to migrate to this package, but we need PHP 7.4 support there.

File: tests/format.txt [hunk 1]
```diff
@@ -976,3 +976,35 @@
...
+  WHEN 3 THEN
+    CASE age
+    WHEN 10 THEN
+      30
+    ELSE
+      31
+    END
+  ELSE
+    40
+  END;
```
Comment (derrabus) [inline tests/format.txt]: Can you please add tests with…

* … a `FROM` clause?
* … a column without a `CASE` statement after the one with the `CASE` statement?
* … a column alias with `AS` for one of the columns built with a `CASE` statement?

Comment (mvorisek) [reply tests/format.txt]: >     * … a `FROM` clause?

do you mean select case ... end from ...?

>     * … a column without a `CASE` statement after the one with the `CASE` statement?

what exactly do you mean?

>     * … a column alias with `AS` for one of the columns built with a `CASE` statement?


do you mean select case ... end AS xxx?

Comment (mvorisek) [reply tests/format.txt]: @derrabus I will do it, but let me please know what you want.

Comment (derrabus) [reply tests/format.txt]: You understood me correctly already. 🙂 



> … a column without a `CASE` statement after the one with the `CASE` statement?

something like this:

```sql
SELECT some_column, CASE … END, some_other_column FROM some_table;
```

Comment (mvorisek) [reply tests/format.txt]: done and retargeted

Comment (greg0ire) [reply tests/format.txt]: Needs a rebase as well

Comment (mvorisek) [reply tests/format.txt]: rebased

**Review by greg0ire — DISMISSED** [review]
To be squash-merged

**Comment by greg0ire** [comment]
:thinking: why are you targeting 1.4.x? This is not a bugfix, is it?

**Comment by mvorisek** [comment]
There is no 1.5.x, if you will create it, feel free to retarget :)

**Review by SenseException — APPROVED** [review]

**Review by greg0ire — APPROVED** [review]

**Comment by greg0ire** [comment]
Thanks @mvorisek !

**Pull request merged**